src/lib/mathlib/math/filter/NotchFilter.hpp

```diff
@@ -165,4 +210,27 @@ T NotchFilter<T>::reset(const T &sample)
 	return apply(sample);
 }
 
+/**
+ * Allows to dynamically update the filtered frequency, refresh rate and quality factor.
+ * Changes the filter's coefficients while conserving the filter's history
+ * Note: This method will only work if using the applyDF1 function (Direct Form I).
+ */
+template<typename T>
+void NotchFilter<T>::update(float sample_freq, float notch_freq, float bandwidth)
+{
+	// backup state
+	T delay_element_1_bkup = _delay_element_1;
```

Comment:
Maybe we could gain a bit of efficiency if we move the zeroing of the delay elements out of `setParameters` so we don't need to backup and restore them. Then you can directly use `setParameters` and we call the new `reset()` function when needed.

Reply:
That sounds good, then the update() method is redundant and just would be using the setParameters() method to update the filters when needed. In that case, I guess we can just initialise the delay elements directly in the class definition and it should do the job just fine. Is that what you meant?

Concerning the reset() method, I guess I should update it with
`_delay_element_output_1 = {};`
`_delay_element_output_2 = {};`
to make sure those are zeroed too, if my understanding of the method is correct.

Reply:
Hi @bresch , I have made changes according to your review, following the approach outlined my comment above. You can refer to a more detailed description within the commit message.

Not all of the CI tests are passing, with the Offboard position control test failing. However I believe it is a problem with the test itself as I haven't modified code relevant to that test.

Let me know if there are any further changes required. Thanks!